k4e-agent-sti.sh

```diff
@@ -133,7 +133,7 @@ createrepo /var/www/html/k4e-repo/
 #-----------------------------------
 git clone https://github.com/ydayagi/r4e.git /home/builder/r4e
 cd /home/builder/r4e
-./r4e-image.sh -s $(hostname) -i $IMAGE_NAME -o $HTTP_API -p http://$(hostname)/k4e-repo/
+./r4e-image.sh -s $(hostname -I) -i $IMAGE_NAME -o $HTTP_API -p http://$(hostname -I)/k4e-repo/
```

Comment:
```
→ hostname -I
10.100.102.22 192.168.122.1 192.168.130.1 192.168.111.1 192.168.42.1 172.17.0.1 192.168.49.1 10.35.206.26 
```
multiple addresses will break this

Reply:
it would be safer to add an argument for that and fallback to this default if haven't provided

Reply:
Better not fall back to a default that may fail (i.e. in case of multiple addresses). Why not add a mandatory argument and fail if not provided in a deterministic way ?

Reply:
added

Reply:
the default (current) behavior of `$(hostname)` should return only the host name.
the problem is with vms that don't have their name resolvable (e.g local vms) and the address might not be reachable outside of their hypervisor (if not configured specifically for that).

if the vmbuilder machine doesn't have a known dns name, nor an accessible ip address,  then the user should provide one, else, the edgedevice will fail to connect to it.

before forcing another parameter, we can add a check for the hostname to be resolvable (either by the parameter or by checking the hostname).

One comment about the ISO though, using the ISO to start edgedevice shouldn't make attempt to access the http-server with the commit, since the ISO already contains the data in it.
Using ks file to point to an external http-server while using the ISO to boot a device isn't the right way to go.
Only the ISO should be used to boot the edgedevice, without any external ks.